bump_dictionary/models/legacy_dictionary_model.py

```diff
@@ -0,0 +1,164 @@
+    BaseModel,
+    ConfigDict,
+    Field,
+    RootModel,
+)
+from pydantic_core import PydanticCustomError
+from typing_extensions import Annotated
+
+
+def validate_unique_list(values: List[str]) -> List[str]:
+    """
+    Check that provided list only has unique elements.
+
+    This custom validator is needed because constrained dtypes and their `unique_items` parameter
+    were deprecated in Pydantic v2. This function was adapted from https://github.com/pydantic/pydantic-core/pull/820#issuecomment-1656228704
+    and https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators.
+
+    See also:
+    - https://docs.pydantic.dev/latest/migration/#changes-to-pydanticfield
+    - https://docs.pydantic.dev/latest/api/types/#pydantic.types.conlist)
+    """
+    if len(values) != len(set(values)):
+        raise PydanticCustomError(
+            "unique_list", f"{values} is not a unique list"
+        )
+    return values
+
+
+class Identifier(BaseModel):
+    """An identifier of a controlled term with an IRI"""
+
+    termURL: str = Field(
+        ...,
+        description="An unambiguous identifier for the term, concept or entity that is referenced",
+        alias="TermURL",
+    )
+    label: str = Field(
+        ...,
+        description="A human readable label. If more than one label exists for the term, "
+        "then the preferred label should be used.",
+        alias="Label",
+    )
+
+
+class Neurobagel(BaseModel):
+    """The base model for a Neurobagel column annotation"""
+
+    isAbout: Identifier = Field(
+        ...,
+        description="The concept or controlled term that describes this column",
+        alias="IsAbout",
+    )
+    missingValues: Annotated[
+        List[str],
+        AfterValidator(validate_unique_list),
+        Field(
+            [],
+            description="A list of unique values that represent "
+            "invalid responses, typos, or missing data",
+            alias="MissingValues",
+            json_schema_extra={"uniqueItems": True},
+        ),
+    ]
+
+    model_config = ConfigDict(extra="forbid")
+
+
+class CategoricalNeurobagel(Neurobagel):
+    """A Neurobagel annotation for a categorical column"""
+
+    levels: Dict[str, Identifier] = Field(
+        ...,
+        description="For categorical variables: "
+        "An object of values (keys) in the column and the semantic "
+        "term (URI and label) they are unambiguously mapped to.",
+        alias="Levels",
+    )
+
+
+class ContinuousNeurobagel(Neurobagel):
+    """A Neurobagel annotation for a continuous column"""
+
+    format: Identifier = Field(
+        ...,
+        description="For continuous columns this field is used to describe "
+        "the format of the raw numerical values in the column. This information is used to transform "
+        "the column values into the desired format of the standardized "
+        "data element referenced in the IsAbout attribute.",
+        alias="Format",
+    )
```

Comment:
I'd just add "Transformation" here as well - with a warning that this is a local workaround